drivers/bluetooth/hci/h4.c

```diff
@@ -135,7 +135,7 @@ static inline void get_iso_hdr(void)
 	if (!rx.remaining) {
 		struct bt_hci_iso_hdr *hdr = &rx.iso;
 
-		rx.remaining = sys_le16_to_cpu(hdr->len);
+		rx.remaining = sys_le16_to_cpu(bt_iso_hdr_len(hdr->len));
```

Comment:
It is important to have the 16-bits in machine endian-ness before applying the 14-bit mask.

```suggestion
		rx.remaining = bt_iso_hdr_len(sys_le16_to_cpu(hdr->len));
```

Reply:
You are absolutely correct - Somehow my mind went to "we must apply the Mask before converting to CPU endianess as we are applying a LE mask" but of course the endianess of the macro is always CPU endian and not LE endian... Thanks!